src/Illuminate/Console/Scheduling/Schedule.php

```diff
@@ -255,9 +275,39 @@ public function exec($command, array $parameters = [])
 
         $this->events[] = $event = new Event($this->eventMutex, $command, $this->timezone);
 
+        $this->mergeAttributes($event);
+
         return $event;
     }
 
+    /**
+     * Create new schedule group.
+     */
+    public function group(Closure $events): void
+    {
+        $this->groupStack[] = $this->attributes;
+
+        $events($this);
+
+        array_pop($this->groupStack);
+    }
+
+    /**
+     * Merge the attributes with the given event.
+     */
+    protected function mergeAttributes(Event $event): void
+    {
+        if (isset($this->attributes)) {
+            $this->attributes->mergeAttributes($event);
+            unset($this->attributes);
+        }
+
+        if (! empty($this->groupStack)) {
+            $group = end($this->groupStack);
+            $group->mergeAttributes($event);
```

Comment:
Nitpick -- unnecessary variable creation:

```suggestion
            end($this->groupStack)->mergeAttributes($event);
```

Reply:
Thanks! But I prefer the current approach for clarity and debugging, and it’s consistent with similar usage elsewhere in the framework.

Reply:
Ok sounds good!